src/platforms/ContractCall/ContractCallCardMenu.tsx

```diff
@@ -17,22 +18,38 @@ const ContractCallCardMenu = ({ platformGuildId }: Props): JSX.Element => {
   )
 
   const { isOpen, onOpen, onClose } = useDisclosure()
+  const {
+    isOpen: legacyIsOpen,
+    onOpen: legacyOnOpen,
+    onClose: legacyOnClose,
+  } = useDisclosure()
+
+  const isLegacy =
+    guildPlatform.platformGuildData.function ===
+    ContractCallFunction.DEPRECATED_SIMPLE_CLAIM
 
   return (
     <>
       <PlatformCardMenu>
-        <MenuItem icon={<PencilSimple />} onClick={onOpen}>
-          Edit NFT description
+        <MenuItem icon={<PencilSimple />} onClick={isLegacy ? legacyOnOpen : onOpen}>
+          {isLegacy ? "Edit NFT description" : "Edit NFT"}
         </MenuItem>
-        <EditRewardAvailabilityMenuItem platformGuildId={platformGuildId} />
         <RemovePlatformMenuItem platformGuildId={platformGuildId} />
       </PlatformCardMenu>
 
-      <EditNFTDescriptionModal
-        isOpen={isOpen}
-        onClose={onClose}
-        guildPlatform={guildPlatform}
-      />
+      {isLegacy ? (
+        <EditNFTDescriptionModal
```

Comment:
We could technically re-use the new Edit NFT modal here, but that would require more internal logic in that component, so I added a ternary here instead. We'll probably drop support for older NFT rewards eventually, we'll be able to remove this logic then.